src/spec/controllers/tags_controller_spec.rb

```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+RSpec.describe TagsController do
+  describe 'GET #show' do
+    let!(:functions_count) { 3 }
+
+    let(:tag) { create :tag }
+    let(:functions) { create_list :function, functions_count, tags: [tag] }
+
+    it 'returns all the funtions associated with this tag' do
+      get :show, params: { id: tag }
+      expect(assigns(:functions)).to eq(functions)
```

Comment:
I see we added a gem for `assigns` I really prefer not to add this gem (or any gem in general), so is there any other way to test what you need without `assigns` ?

Reply:
I can test the HTTP status returned from the request but it doesn't reflect much about the functionality of the system. I have made some researches and I found that the recommended way to test the returned values from the controller is to test the DOM itself through a feature test which I couldn't find any of it in the project. My final conclusion is I'll remove the gem with the test (making it only checks the returned HTTP status) but keeping in my mind that we should implement feature tests in the project from now on for more accuracy.